Service/InstallationService.php

```diff
@@ -193,11 +193,23 @@ private function createEndpoints($objectsThatShouldHaveEndpoints): array
         return $endpoints;
     }
 
+    public function setEntityMaxDepth()
+    {
+        $entities = $this->entityManager->getRepository('App:Entity')->findAll();
+        foreach ($entities as $entity) {
+            // set maxDepth for an entity to 5
+            $entity->setMaxDepth(5);
+            $this->entityManager->persist($entity);
+        }
+    }
+
     private function addSchemasToCollection(CollectionEntity $collection, string $schemaPrefix): CollectionEntity
     {
         $entities = $this->entityManager->getRepository('App:Entity')->findByReferencePrefix($schemaPrefix);
         foreach ($entities as $entity) {
             $entity->addCollection($collection);
+            // set all entity maxDepth to 5
+            $this->setEntityMaxDepth();
```

Comment:
If the setEntityMaxDepth function loops through all entities, wouldn't it be better to run the function after all entities have been added, instead of running it every time an entity is added to a collection?